internal/hcs/waithelper.go

```diff
@@ -17,6 +17,11 @@ func processAsyncHcsResult(err error, resultp *uint16, callbackNumber uintptr, e
 
 func waitForNotification(callbackNumber uintptr, expectedNotification hcsNotification, timeout *time.Duration) error {
 	callbackMapLock.RLock()
+	if _, ok := callbackMap[callbackNumber]; !ok {
+		callbackMapLock.RUnlock()
+		logrus.Errorf("failed to waitForNotification: callbackNumber %d does not exist in callbackMap", callbackNumber)
+		return ErrInvalidCallback
```

Comment:
I think this is unfortunately a breaking change to v1

Reply:
Maybe you could check if its `ErrInvalidCallback` and convert it for the v1 entrypoint?

Reply:
It's not really a breaking change - the daemon would have hit a panic, never having returned any error

Reply:
Updated to ErrHandleClose